Automattic/team51-cli
# Move SFTP connection functionality into API_Helper

Moving the SFTP connection functionality into the `API_Helper` class so it can be used in #100 .

Changes in this PR:
- [Move SFTP connection functionality into API_Helper](https://github.com/Automattic/team51-cli/commit/ccd5d910b6702806b0637e976e467212da509242)
- In `pressable_call_api`, [encode JSON data before passing to Helper](https://github.com/Automattic/team51-cli/commit/46f00ddabfc6181385ea7c48b145a45692960f2a)

Testing:
- Run `php-errors` on a site that already has `concierge+cli@a8c.com` as a collaborator
- Run `php-errors` on a site that doesn't have `concierge+cli@a8c.com` as a collaborator
- Make sure both commands ran properly
- Call the `pressable_call_api` command with some encoded data. Example: `team51 pressable-call-api --query='sites/1247837/collaborators' --method=POST --data='{"siteID":1247837,"email":"concierge+test.account@a8c.com"}'`
- Make sure the test command ran properly (the above command creates a collaborator account on [mikestraw-test-production](https://my.pressable.com/sites/1247837)
- Please test around this issue
